daemon/swap_daemon_erc20_test.go

```diff
@@ -21,26 +24,52 @@ import (
 // Tests the scenario where Bob has XMR and enough ETH to pay gas fees for the token claim. He
 // exchanges 2 XMR for 3 of Alice's ERC20 tokens.
 func TestRunSwapDaemon_ExchangesXMRForERC20Tokens(t *testing.T) {
-	minXMR := coins.StrToDecimal("1")
-	maxXMR := coins.StrToDecimal("2")
-	exRate := coins.StrToExchangeRate("1.5")
+	fundingEC := extethclient.CreateTestClient(t, tests.GetTakerTestKey(t))
+	tokenAsset := getMockTetherAsset(t, fundingEC)
+	tokenAddr := tokenAsset.Address()
+	token, err := fundingEC.ERC20Info(context.Background(), tokenAddr)
+	require.NoError(t, err)
+
+	minXMR := coins.StrToDecimal("0.1")
+	maxXMR := coins.StrToDecimal("0.25")
+	exRate := coins.StrToExchangeRate("140")
+	providesAmt := coins.NewEthAssetAmount(coins.StrToDecimal("33.999994"), token) // 33.999994 USDT / 140 = 0.2428571 XMR
+	gasMoney := coins.EtherToWei(coins.StrToDecimal("0.1"))
 
-	bobConf := CreateTestConf(t, tests.GetMakerTestKey(t))
+	bobEthKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+	bobConf := CreateTestConf(t, bobEthKey)
 	monero.MineMinXMRBalance(t, bobConf.MoneroClient, coins.MoneroToPiconero(maxXMR))
 
-	aliceConf := CreateTestConf(t, tests.GetTakerTestKey(t))
+	// Ensure that Alice has no tokens and definitely no pre-approval to spend
+	// any of those tokens by giving her a brand-new ETH key.
+	aliceEthKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+	aliceConf := CreateTestConf(t, aliceEthKey)
+	timeout := 7 * time.Minute
 
-	tokenAsset := getMockTetherAsset(t, aliceConf.EthereumClient)
+	// Fund Alice and Bob with a little ether for gas. Bob needs gas to claim,
+	// as ERC20 token swaps cannot use a relayer.
+	_, err = fundingEC.Transfer(context.Background(), aliceConf.EthereumClient.Address(), gasMoney, nil)
```

Comment:
This line should be updated when we add the new token transfer API to the extended ETH client:
https://github.com/AthanorLabs/atomic-swap/issues/485